exec/java-exec/src/test/java/org/apache/drill/exec/store/log/TestLogReader.java

```diff
@@ -0,0 +1,368 @@
+        .addNullable("field_4", MinorType.VARCHAR)
+        .build();
+
+    RowSet expected = client.rowSetBuilder(expectedSchema)
+        .addRow("070823", "root@localhost on test1")
+        .addRow("070823",  "show tables")
+        .addRow("070823",  "select * from category" )
+        .addRow("070917",  "select * from location" )
+        .addRow("070917", "select * from location where id = 1 LIMIT 1" )
+        .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
+  }
+
+  @Test
+  public void testRawNoSchema() throws RpcException {
+    String sql = "SELECT _raw FROM cp.`regex/mysql.sqllog`";
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+
+    BatchSchema expectedSchema = new SchemaBuilder()
+        .addNullable("_raw", MinorType.VARCHAR)
+        .build();
+
+    RowSet expected = client.rowSetBuilder(expectedSchema)
+        .addRow("070823 21:00:32       1 Connect     root@localhost on test1")
+        .addRow("070823 21:00:48       1 Query       show tables")
+        .addRow("070823 21:00:56       1 Query       select * from category" )
+        .addRow("070917 16:29:01      21 Query       select * from location" )
+        .addRow("070917 16:29:12      21 Query       select * from location where id = 1 LIMIT 1" )
+        .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
+  }
+
+  @Test
+  public void testUMNoSchema() throws RpcException {
+    String sql = "SELECT _unmatched_rows FROM cp.`regex/mysql.sqllog`";
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+
+    BatchSchema expectedSchema = new SchemaBuilder()
+        .addNullable("_unmatched_rows", MinorType.VARCHAR)
+        .build();
+
+    RowSet expected = client.rowSetBuilder(expectedSchema)
+        .addRow("dfadkfjaldkjafsdfjlksdjflksjdlkfjsldkfjslkjl")
+        .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
+  }
+
+  @Test
+  public void testRawUMNoSchema() throws RpcException {
+    String sql = "SELECT _raw, _unmatched_rows FROM cp.`regex/mysql.sqllog`";
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+
+    BatchSchema expectedSchema = new SchemaBuilder()
+        .addNullable("_raw", MinorType.VARCHAR)
+        .addNullable("_unmatched_rows", MinorType.VARCHAR)
+        .build();
+
+    RowSet expected = client.rowSetBuilder(expectedSchema)
+        .addRow("070823 21:00:32       1 Connect     root@localhost on test1", null)
+        .addRow("070823 21:00:48       1 Query       show tables", null)
+        .addRow("070823 21:00:56       1 Query       select * from category", null )
+        .addRow("070917 16:29:01      21 Query       select * from location", null )
+        .addRow("070917 16:29:12      21 Query       select * from location where id = 1 LIMIT 1", null )
+        .addRow( null, "dfadkfjaldkjafsdfjlksdjflksjdlkfjsldkfjslkjl")
+        .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
+  }
+
+}
```

Comment:
Missing a test for a bogus type name, repeated field names and other similar error conditions. What else should be checked? Null or empty file extension? Negative minimum errors? min/max errors out of sync? What else?

The way to think about this is: we only need to test those cases that we don't want the Drill team to have to fix in response to an escalation in a production system. If it is OK to wait to fix the issue until an escalation, we don't need to test it. (Now, I'm sure the Drill team will suggest that there are no such cases that are OK to fix in an escalation...)

Here's how I think about this. For every field that that the user might set, think of all possible values, write code to test for, and fail, for bad values, then write tests to provide that he bad values are rejected.